All_home_works/hw3/task2.py

```diff
@@ -0,0 +1,47 @@
+"""
+Here's a not very efficient calculation function that calculates something important:
+
+import time
+import struct
+import random
+import hashlib
+
+def slow_calculate(value):
+
+    time.sleep(random.randint(1,3))
+    data = hashlib.md5(str(value).encode()).digest()
+    return sum(struct.unpack('<' + 'B' * len(data), data))
+
+Calculate total sum of slow_calculate() of all numbers starting from 0 to 500.
+ Calculation time should not take more than a minute.
+ Use functional capabilities of multiprocessing module.
+  You are not allowed to modify slow_calculate function.
+"""
+import hashlib
+import random
+import struct
+import time
+from multiprocessing import Pool
+
+
+def slow_calculate(value):
+    time.sleep(random.randint(1, 3))  # noqa: S311
+    data = hashlib.md5(str(value).encode()).digest()  # noqa: S303
+    return sum(struct.unpack("<" + "B" * len(data), data))
+
+
+def timer(func):
+    def wrapper(*args, **kwargs):
+        start = time.time()
+        result = func(*args, **kwargs)
+        delta = time.time() - start
+        return result, delta
+
+    return wrapper
+
+
+@timer
+def sum_of_slow_calculate(max_value=500):
+    pool = Pool(processes=max_value)
```

Comment:
Is it a recommended way to use Pool?

Reply:
Nope. I forgot about the context manager.
By the way, I have a question. Maybe better to use threading if we have 500 processes?

Reply:
Threading will not work, you will get an explanation of why in the corresponding lecture.

Why do we need to have 500 processes?

Reply:
It's actually not needed. But, I thought, in multiprocessing, it's recommended to use such many processes as we have cores in our system.

Reply:
>  it's recommended to use such many processes as we have cores in our system.

It's ok to have 500 cores on your machine. But GitHub actions are run on less powerful machines, let's set more reasonable numbers. 

As well please do some experiments and check the number of processes that will give the best performance on your machine (binary search, 1 execution per number, we don't need high accuracy)